cylc/flow/task_proxy.py

```diff
@@ -355,25 +355,38 @@ def get_point_as_seconds(self):
                 self.point_as_seconds += utc_offset_in_seconds
         return self.point_as_seconds
 
-    def get_clock_trigger_time(self, offset_str):
-        """Compute, cache, and return trigger time relative to cycle point.
+    def get_clock_trigger_time(
+        self,
+        point: 'PointBase', offset_str: Optional[str] = None
+    ) -> int:
+        """Compute, cache and return trigger time relative to cycle point.
 
         Args:
-            offset_str: ISO8601Interval string, e.g. "PT2M".
-                        Can be None for zero offset.
+            point:
+                String representing itask string.
+            offset_str:
+                ISO8601Interval string, e.g. "PT2M".
+                Can be None for zero offset.
         Returns:
             Absolute trigger time in seconds since Unix epoch.
 
         """
-        if self.clock_trigger_time is None:
-            if offset_str is None:
-                trigger_time = self.point
+        # None cannot be used as a dict key:
+        offset_str = offset_str if offset_str else 'no-offset'
+
+        if (
+            not self.clock_trigger_times
+            or offset_str not in self.clock_trigger_times
+        ):
+            if offset_str == 'no-offset':
```

Comment:
Can use any zero duration here, we tend to use `P0Y`:

```suggestion
            if offset_str == 'P0Y':
```